devel/fortio/logger.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fortio
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"runtime"
+	"strings"
+)
+
+// LogLevel is the level of logging (0 Debug -> 6 Fatal).
+type LogLevel int
+
+// Log levels
+const (
+	D LogLevel = iota // Debug
+	V LogLevel = iota // Verbose
+	I LogLevel = iota // Info
+	W LogLevel = iota // Warning
+	E LogLevel = iota // Error
+	C LogLevel = iota // Critical
+	F LogLevel = iota // Fatal
+)
+
+var (
+	level          = I // default is Info and up
+	levelToStrA    []string
+	levelToStrM    map[string]LogLevel
+	logPrefix      = flag.String("logprefix", "> ", "Prefix to log lines before logged messages")
```

Comment:
how do you reuse flags that are in a main ? glog seems to have their flag in the lib too